### PR TITLE
Added more tests to main

### PR DESCRIPTION
**Pull request by nasaownsky**
## Description of the change

Added more tests to main

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

> Closes [#XXXX]

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: src/utils/__tests__/generateSourceText.test.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+// =============================================================================
+import generateSourceText from "../generateSourceText";
+
+describe("generateKeyInsightsData.js", () => {
```
Comment (jessex) [inline src/utils/__tests__/generateSourceText.test.js:19]: Should be `generateSourceText.js`

File: src/utils/__tests__/generateKeyInsightsData.test.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+// =============================================================================
+import generateKeyInsightsData from "../generateKeyInsightsData";
+
+describe("generateKeyInsightsData.js", () => {
```
Comment (jessex) [inline src/utils/__tests__/generateKeyInsightsData.test.js:19]: Are you planning to add more test cases for the other functions in this file that are not tested here yet?

**Review by jessex — COMMENTED** [review]
Looking forward to a nice bump in test coverage, @nasaownsky! Do you have more tests in this particular PR planned, or is this it?

**Comment by nasaownsky** [comment]
For now I'm focused on #80, but I can add few more tests if I don't run into any difficulties with it.

File: src/utils/__tests__/generateSourceText.test.js (hunk 1 above)
Comment (nasaownsky) [reply src/utils/__tests__/generateSourceText.test.js:19]: Fixed

File: src/utils/__tests__/generateKeyInsightsData.test.js (hunk 2 above)
Comment (nasaownsky) [reply src/utils/__tests__/generateKeyInsightsData.test.js:19]: Yes. But maybe we should at first decide what to do with `v2-jails`?

**Comment by nasaownsky** [comment]
@jessex I've covered `generateKeyInsightsData` with tests in #83, because it has some changes there.

**Review by jessex — APPROVED** [review]
Fantastic!

**Pull request merged**